src/main/java/jenkins/plugins/nodejs/tools/MirrorNodeJSInstaller.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @DataBoundSetter
+    public void setMirrorURL(String mirrorURL) {
+        this.mirrorURL = mirrorURL;
+    }
+
+    public String getMirrorURL() {
+        return this.mirrorURL;
+    }
+
+    @DataBoundSetter
+    public void setSkipTLSVerify(boolean skipTLSVerify) {
+        this.skipTLSVerify = skipTLSVerify;
+    }
+
+    public boolean isSkipTLSVerify() {
+        return this.skipTLSVerify;
+    }
+
+    @DataBoundSetter
+    public void setCredentialsId(String credentialsId) {
+        this.credentialsId = credentialsId;
+    }
+
+    public String getCredentialsId() {
+        return this.credentialsId;
+    }
+
+    TrustManager[] trustAllCerts = new TrustManager[]{
+            new X509TrustManager() {
+                public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                    return new X509Certificate[0];
+                }
+
+                public void checkClientTrusted(
+                        java.security.cert.X509Certificate[] certs, String authType) {
+                }
+
+                public void checkServerTrusted(
+                        java.security.cert.X509Certificate[] certs, String authType) {
+                }
+            }
+    };
+
+    @Override
+    public FilePath performInstallation(ToolInstallation tool, Node node, TaskListener log) throws IOException, InterruptedException {
+        if (this.isSkipTLSVerify()) {
+            try {
+                SSLContext sc = SSLContext.getInstance("SSL");
+                sc.init(null, trustAllCerts, new java.security.SecureRandom());
+                HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+            } catch (KeyManagementException | NoSuchAlgorithmException e) {
+                e.printStackTrace();
+            }
+        }
+        return super.performInstallation(tool, node, log);
+    }
+
+    @Override
+    public Installable getInstallable() throws IOException {
+        Installable installable = super.getInstallable();
+        return installable != null ? new MirrorNodeJSInstallable(installable) : installable;
+    }
+
+    protected final class MirrorNodeJSInstallable extends NodeSpecificInstallable {
+
+        public MirrorNodeJSInstallable(Installable inst) {
+            super(inst);
+        }
+
+        @Override
+        public NodeSpecificInstallable forNode(Node node, TaskListener log) throws IOException {
+            InstallerPathResolver installerPathResolver = InstallerPathResolver.Factory.findResolverFor(id);
+            String relativeDownloadPath = installerPathResolver.resolvePathFor(id, ToolsUtils.getPlatform(node), ToolsUtils.getCPU(node));
+            String baseURL = url.replace(PUBLIC_NODEJS_URL, mirrorURL);
+            url = baseURL + relativeDownloadPath;
+            return this;
+        }
+
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends NodeJSInstaller.DescriptorImpl { // NOSONAR
```

Comment:
this descriptor must be typed,. What I mean in my last comment is extends an (new) abstract class parametric in T that have the 3 common methods.